enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,364 @@
+- The priority (order) of the `nftables` entries relative to other services should be coordinated with the OpenShift networking team to ensure it follows established precedent and does not interfere with other networking rules.
+
+#### nftables Presence and Management
+
+- `nftables` is always present as part of the RHCOS payload. Baremetal and AWS (at least) OCP clusters install `nftables` tables, chains, and rules by default.
+- The `nftables` systemd unit is disabled by default, but the netfilter subsystem is active and can be configured via the `nft` CLI/API without enabling the systemd unit.
+- OVN-Kubernetes relies on netfilter (iptables/nftables) for features like UDN, Egress, and Services.
+- The Machine Config Operator (MCO) does not manage `nftables` directly, but users can explicitly disable or modify `nftables` via MachineConfig if desired.
+- If a user disables or removes `nftables`, this is considered an explicit user-driven action and is not managed or expected by OpenShift.
+- If `nftables` is not present or is disabled, the proxy will not function as intended. Detection of this condition should be implemented (e.g., by checking for the presence of the `nft` binary and ability to apply rules), and the operator should surface a clear error or degraded status.
+- Based on current RHCOS and OCP design, it is not possible to remove the underlying netfilter subsystem, so the feature can reliably depend on its presence unless a user takes explicit unsupported action.
+
+### Design Details
+
+- **Operator Integration**: The cluster-kube-apiserver-operator will manage the complete lifecycle of the HTTP01 challenge proxy, including deployment, configuration, upgrades, and removal.
+- **Proxy Deployment**: The operator will deploy the proxy feature as a DaemonSet on control plane nodes. The daemonset will implement nftable rules via pods that run to completion.
+- **Traffic Redirection**: This will use `nftables` rules to redirect incoming traffic on `API:80` to `PROXY:8888`.
+- **Security**: The proxy implements selective traffic filtering, only handling HTTP requests with paths matching `/.well-known/acme-challenge/*` destined for the API VIP. 
+  - **ACME Traffic**: Requests to `<API_VIP>/.well-known/acme-challenge/*` are redirected to the ingress routers for certificate validation.
+    - **Valid ACME Data**: Properly formatted challenge requests are forwarded to cert-manager challenge pods for validation.
+    - **Invalid ACME Data**: Malformed challenge requests to the correct path are still forwarded to cert-manager, which handles the validation failure (same behavior as without the proxy).
+  - **Non-ACME Traffic**: HTTP requests to the API VIP on port 80 with other paths receive a **403 Forbidden** response with the message "Only /.well-known/acme-challenge/* is allowed".
```

Comment:
Would a 400 Bad Request be more appropriate than a 403 Forbidden?

A 403 makes it sound like it is an explicit restriction of permissions for an individual user as opposed to the proxy only understanding how to handle traffic to the `/.well-known/acme-challenge/*` endpoint.

Reply:
Updated in the doc.